ls1intum/Artemis
# `Exam mode`: Clean up old participations when regenerating student exams

### Checklist
#### General


#### Server
#### Client


### Motivation and Context
This is a fix for an issue reported by @S-Eibl: 'When regenerating an exam the participations are not deleted for students in an exercise group'.
Currently, regenerating student exams would create new participations, but did not remove the old one.
This could lead to a issues where e.g. 3 students have 4 exercises allocated to them:
![image](https://user-images.githubusercontent.com/5084100/180601179-7a473abc-142c-497a-a6ff-6acbb03ddf5f.png)


### Description
The exam gets reset when ALL student exams are regenerated.
This behavior is now also mentioned in the corresponding modal.

### Steps for Testing
1. Create a new exam with 1 exercise group and 2 exercises in it
2. Register a student (or multiple ones) to the exam
3. Generate individual exams
4. Prepare exercise start
5. Check the overview: the students should be allocated to exercises in some way
![image](https://user-images.githubusercontent.com/5084100/180600596-66359931-ce30-4ca5-ace0-e93067df83ab.png)
6. Generate all individual exams again (not only missing ones)
7. Prepare exercise start
8. Check the overview again: the students should be allocated to exercises in some way, which might be different from the one in step 5. The total number of participants should be equal to the number of students
![image](https://user-images.githubusercontent.com/5084100/180600634-889c4ca3-7f3b-4f81-ada6-a064ea39299b.png)
**Note:** Due to the randomness in the exercise allocation, you might have to repeat steps 6-8 a few times


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots
![image](https://user-images.githubusercontent.com/5084100/180601467-5457ad93-c1b8-410d-9a15-b090d40c5cf0.png)
